### PR TITLE
[CLOUD-2647] AMQ broker pod blocked when split-x folder is locked by A…

### PR DESCRIPTION
**Pull request by RoddieKieley**
…MQ Drainer pod

* When terminating changed the flock -w, i.e. waiting, to flock -n, no waiting
* Updated the log message for the terminating case to indicate no waiting
* https://issues.jboss.org/browse/CLOUD-2647

Signed-off-by: Roddie Kieley <rkieley@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets following requirements:

- [X] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [X] Pull Request contains link to the JIRA issue
- [X] Pull Request contains description of the issue
- [X] Pull request does not include fixes for other issues than the main ticket
- [X] Attached commits represent unit of work and are properly formatted
- [X] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [X] Every commit contains `Signed-off-by: Your Name <yourname@redhat.com>` - use `git commit -s`

### CODE REVIEW

File: os-partition/added/partitionPV.sh [hunk 1]
```diff
@@ -34,9 +34,24 @@ function partitionPV() {
...
           else
-            # Terminating, grab the lock with timeout
-            echo "Existing server instance is terminating, waiting to acquire the lock"
-            flock -w $LOCK_TIMEOUT $LOCK_FD
+            REMAINING_LOCKING_TIME=$LOCK_TIMEOUT
+            echo "Existing server instance is terminating, attempting lock per second for $LOCK_TIMEOUT seconds"
+
+            while : ; do
+              flock -n $LOCK_FD
+              if [$? -eq 0 ]; then
```
Comment (Crumby) [inline os-partition/added/partitionPV.sh]: `if [ $? -eq 0 ]; then` -> missing space after [.

File: os-partition/added/partitionPV.sh [hunk 2]
```diff
@@ -34,9 +34,24 @@ function partitionPV() {
...
+                # Grabbed the lock successfully
+                break
+              fi
+              $REMAINING_LOCKING_TIME=$(expr $REMAINING_LOCKING_TIME - 1)
```
Comment (Crumby) [inline os-partition/added/partitionPV.sh]: I believe that we want to assign to `REMAINING_LOCKING_TIME`. $ shouldn't be there.

File: os-partition/added/partitionPV.sh (hunk 1 above)
Comment (RoddieKieley) [reply os-partition/added/partitionPV.sh]: @Crumby Thanks for catching that, addressed!

File: os-partition/added/partitionPV.sh (hunk 2 above)
Comment (RoddieKieley) [reply os-partition/added/partitionPV.sh]: @Crumby Correct, thanks, addressed below.

File: os-partition/added/partitionPV.sh [hunk 3]
```diff
@@ -33,12 +33,14 @@ function partitionPV() {
...
             # Not terminating, grab the lock without waiting
             flock -n $LOCK_FD
+            LOCK_STATUS=$?
           else
             REMAINING_LOCKING_TIME=$LOCK_TIMEOUT
             echo "Existing server instance is terminating, attempting lock per second for $LOCK_TIMEOUT seconds"
 
             while : ; do
               flock -n $LOCK_FD
+              LOCK_STATUS=$?
```
[the remark this answers is not in the document]
Comment (RoddieKieley) [reply os-partition/added/partitionPV.sh]: @Crumby Updated to use the now set $LOCK_STATUS instead of the value that was just used and reset in $?

**Pull request merged**